### PR TITLE
Aux loss is already included in the loss returned by Transformers

### LINKED ISSUE #4070: MOE aux loss bug
### Reproduction

I saw that trl added the aux loss for moe calculation:
```
        # Add auxiliary loss if available
        if self.aux_loss_enabled and self.aux_loss_coef:
            aux_loss = self.aux_loss_coef * outputs.aux_loss
            loss += aux_loss
```

However, in the implementation of qwen3 moe in the Transformer, it can be seen that the final loss has already added the aux_loss.
https://github.com/huggingface/transformers/blob/31ab7168ff7e07f61c90134e5238c4d97606aa70/src/transformers/models/qwen3_moe/modeling_qwen3_moe.py#L686

So, is the aux loss added repeatedly to the loss in trl?

### System Info

trl v0.23.0

### Checklist

- [x] I have checked that my issue isn't already filed (see [open issues](https://github.com/huggingface/trl/issues?q=is%3Aissue))
- [x] I have included my system information
- [x] Any code provided is minimal, complete, and reproducible ([more on MREs](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [x] Any code provided is properly formatted in code blocks, (no screenshot, [more on code blocks](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [x] Any traceback provided is complete

### PR DESCRIPTION
**Pull request by pramodith**
# What does this PR do?

Seems like the Transformer's lib already accounts for the `aux_loss` in MoE models by adding it to the final returned loss in the `forward` function of each model's implementation. So we revert the changes in #4012, the doc strings added in the PR are still valid so they'll remain.

see the issue for more info.

Fixes #4070 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/huggingface/trl/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [X] Was this discussed/approved via a GitHub issue? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: trl/trainer/sft_trainer.py [hunk 1]
```diff
@@ -1177,9 +1162,6 @@ def compute_loss(self, model, inputs, return_outputs=False, num_items_in_batch=N
                 total_sum = total_tokens.sum()
                 accuracy = (correct_tokens.sum() / total_sum).item() if total_sum > 0 else 0.0
                 self._metrics[mode]["mean_token_accuracy"].append(accuracy)
-                if self.aux_loss_enabled:
-                    aux_loss = self.accelerator.gather_for_metrics(aux_loss).mean().item()
```
Comment (qgallouedec) [inline trl/trainer/sft_trainer.py:1181]: maybe we can still log it. It should be included in the output: https://github.com/huggingface/transformers/blob/31ab7168ff7e07f61c90134e5238c4d97606aa70/src/transformers/models/qwen3_moe/modeling_qwen3_moe.py#L690

Comment (qgallouedec) [reply trl/trainer/sft_trainer.py:1181]: something like 

```python
if isinstance(output, MoeCausalLMOutputWithPast):
   ...
```

Comment (pramodith) [reply trl/trainer/sft_trainer.py:1181]: In that case, I think retaining the test case is a good idea too.

**Review by qgallouedec — APPROVED** [review]
Thanks for the fix!
